### PR TITLE
Update active db size calculation to use only leaf nodes

### PR DESCRIPTION
**Pull request by nickva**
Previously it used both leaf and intermediate nodes.

With this PR active db size should decrease when users delete documents. This, in turn, should make smoosh enqueue those dbs for compaction to recover the disk space used by the now deleted document bodies. Previously, it was possible for users to delete gigabytes worth of document bodies and smoosh never noticing and never triggering the compaction.

Original idea and patch provided by Robert Newson in CouchDB dev Slack discussion channel.

Co-authored-by: Robert Newson

Fixes: https://github.com/apache/couchdb/issues/4263

### CODE REVIEW

**Review by janl — APPROVED** [review]
nicely done

**Comment by davisp** [comment]
I think this is papering over a different bug/misbehavior. I'm not sure if we documented it as such, but "active size" should be a rough representation of how many bytes are reachable from the root nodes in the header. The idea there being that active size would be a rough approximation of the size of the database just after compaction finishes.

Going back to the original issue of "deleting gigabytes of data and smoosh not noticing" my wild guess is that given small enough individual documents, coupled with a randomish (with respect to doc_id or update_seq order) could lead to enough "active garbage inner nodes" that smoosh gets confused.

Also, how much data are we missing here? And/or can we generate a test case that demonstrates the behavior? If anyone has run this in production, I'd be curious to see how many compactions are running on the regular, along with the expected vs actual post compaction file sizes.

For instance, in the "smoosh didn't notice and perform compaction", did there end up being a massive file size shrinkage? Or is this is "huh, weird that there's no compaction triggered" issue. Given that deletions are only removing doc bodies and the trees themselves aren't changing shape maybe its just unexpected overhead of the trees being swamped in the smoosh calculations?

Basically, it does sound like there *could* be an issue here, but I doubt that this is the correct fix. Unless I'm forgetting something crazy big, this basically seems like a "external size after compression applied" calculation which might be useful, but is definitely different than what the original intent was.

**Comment by nickva** [comment]
>  small enough individual documents,

There were just over 1Mb and it was about 1000 of them, and smoosh never noticed they were deleted and that it could free almost 1Gb of disk space. There is a gist to try it out https://gist.github.com/nickva/243caf230f7aa3b61f43bbde57063f72

**Comment by davisp** [comment]
Ignore everything I said earlier because I wasn't paying attention enough to realize that the internal nodes not being counted was for key trees, not b trees. This looks right to me in that we shouldn't be counting the size of the deleted internal nodes of a revision tree.

The change in the docs/active size definition should go from something like "The approximate number of bytes to represent this database after compaction" which is slightly different than "The number of active bytes reachable from root" because we're purposefully ignoring bytes we can get to that will no longer exist after compaction.

And that doc change is only if we don't definite it as such already.

Good work and sorry for the Monday morning noise!
